kernel/src/syscall/ioctl.rs

```diff
@@ -39,6 +39,23 @@ pub fn sys_ioctl(fd: FileDesc, cmd: u32, arg: Vaddr, ctx: &Context) -> Result<Sy
             file.set_status_flags(flags)?;
             0
         }
+        IoctlCmd::FIOCLEX => {
+            // Sets the close-on-exec flag of the file.
+            // Follow the implementation of fcntl()
+
+            let flags = FdFlags::CLOEXEC;
+            let file_table = ctx.process.file_table().lock();
+            let entry = file_table.get_entry(fd)?;
+            entry.set_flags(flags);
+            0
+        }
+        IoctlCmd::FIONCLEX => {
+            // Clears the close-on-exec flag of the file.
+            let file_table = ctx.process.file_table().lock();
+            let entry = file_table.get_entry(fd)?;
+            entry.clear_flags();
```

Comment:
The `FIONCLEX` sub-command should only clear `FdFlags::CLOEXEC`. In case that more bits are added to `FdFlags` in the future, the code here should only clear this specific bit.

Reply:
Thank you. Update Already.